File: TimeValidator/TimeValidator.cpp
#include <iostream>
#include <string>

using namespace std;

bool isValidTime(const string& time) {
    if (time.length() != 4) {
        return false; // Должно быть ровно 4 цифры
    }

    // Первые 2 символа - часы, последние 2 символа - минуты
    string hours = time.substr(0, 2);
    string minutes = time.substr(2, 2);

    // Преобразуем строки в числа
    int h = stoi(hours);
    int m = stoi(minutes);

    // Проверяем диапазоны
    return (h >= 0 && h <= 23) && (m >= 0 && m <= 59);
}

int main() {
    string input;
    cout << "Введите последовательность чисел: ";
    cin >> input;

    if (isValidTime(input)) {
        cout << "YES" << endl;
    } else {
        cout << "NO" << endl;
    }

    return 0;
}

File: TimeValidator/TimeValidator.py
def is_valid_time(time):
        # Должно быть ровно 4 цифры
    if len(time) != 4:
        return False
    
    # Первые 2 символа - часы, последние 2 символа - минуты
        hours = time[:2]
minutes = time[2:]

        # Преобразуем строки в числа
h = int(hours)
m = int(minutes)
    
    # Проверяем диапазоны
    return (0 <= h <= 23) and (0 <= m <= 59)

# Запрашиваем ввод у пользователя
input_time = input("Введите последовательность чисел: ")

if is_valid_time(input_time):
print("YES")
else:
print("NO")
